enhancements/sig-policy/89-operator-policy-kind/README.md

```diff
@@ -0,0 +1,557 @@
+    #   matchLabels: # or matchExpressions
+    #     mydomain.io/prod: "true"
+    serviceAccountName: my-og-account # optional
+  subscription:
+    config: {} # optional
+    channel: stable
+    name: my-operator
+    namespace: own-namespace
+    installPlanApproval: Automatic # may be overridden to Manual by other settings
+    source: my-catalog
+    sourceNamespace: my-catalog-namespace
+    startingCSV: my-operator.v0.1.0 # optional
+  allowedCSVs:
+    - my-operator.v0.1.1
+    - my-operator.v0.2.0
+  # - "*" # Allow all
+  removalBehavior:
+    operatorGroups: DeleteIfUnused
+    subscriptions: Delete
+    clusterServiceVersions: Delete
+    installPlans: Keep
+    customResourceDefinitions: Keep
+    apiServiceDefinitions: Keep
+  statusConfig:
+    catalogSourceUnhealthy: Condition
+    deploymentsUnavailable: NonCompliant
+    upgradesAvailable: Condition
+    upgradesProgressing: NonCompliant
+```
+
+When `remediationAction` is set to `inform`, no actions (creates, deletes, updates) will be taken on
+any resources, but the current state may be *read* to identify if the operator is installed as
+specified. When resources are missing, the policy will be NonCompliant, with a message specifying
+what is missing, and that it will not be created because the policy is not being enforced. When it
+is set to `enforce`, OperatorGroups and Subscriptions may be created by the controller, InstallPlans
+may be approved, and resources may be removed when the policy is deleted.
+
+The optional `spec.operatorGroup` section defines an
+[OperatorGroup](https://github.com/operator-framework/api/blob/master/crds/operators.coreos.com_operatorgroups.yaml).
+If not specified, and if the operator supports this InstallMode, then it will ensure that an
+[AllNamespaces](https://olm.operatorframework.io/docs/advanced-tasks/operator-scoping-with-operatorgroups/#targetnamespaces-and-their-relationship-to-installmodes)
+OperatorGroup exists in the namespace of the subscription.
+
+The `spec.subscription` section includes the fields from the spec of an [OLM
+Subscription](https://github.com/operator-framework/api/blob/master/crds/operators.coreos.com_subscriptions.yaml).
+For more information on the `config` field, see:
+https://github.com/operator-framework/api/blob/v0.17.3/pkg/operators/v1alpha1/subscription_types.go#L42.
+The `installPlanApproval` field is optional. If the policy is in `enforce` mode and the allowed CSVs
+are restricted as specified below, then the `installPlanApproval` field on the Subscription will
+always be Manual, regardless of the setting here.
+
+The `spec.allowedCSVs` list specifies what installed versions are considered Compliant when the
+policy is in `inform` mode, and which InstallPlans can be approved when in `enforce` mode. Only
+exact matches are considered, except for a special value, "*" which matches any version. This field
+must be set, and must have at least one item.
```

Comment:
Is there a valid case for this to be set with empty value? If not, is it better to keep it optional and allow all if not set?

Reply:
Good question! I hadn't realized it, but yes, an empty list here could be useful: in particular when the policy is in `inform` mode, that could be used to verify that an operator is *not* installed (similar to `mustnothave` in ConfigurationPolicy).

The design right now doesn't specify whether already-existing subscriptions should be deleted by an OperatorPolicy in `enforce` mode, that might need to be considered...